src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/FtpClient.java

```diff
@@ -80,19 +95,31 @@ public void upload(LocalSourceFile file, boolean isSmokeTestFile) {
                         }
                     })
                     .collect(toList());
+
+                insights.trackFtpReportDownload(Duration.between(now, Instant.now()), true);
```

Comment:
you could combine these last two insights tracking calls by using a finally block after the exception handler, here and in a couple of other places